### ISSUE TITLE
[accessibility] Make the map container a distinct semantic structure that entails all its components

### ISSUE THREAD

**Issue by Malvoz** [maplibre/maplibre-gl-js#364]
When visual users look at a map they can easily tell that the controls, attribution, and other (potentially custom) UI components that are placed on the map belong to it. <span title="Assistive technology">AT</span> users rely on HTML semantics, DOM structure, and ARIA to convey these things and their relationships.

Currently, the `tabindex="0"`, `aria-label="Map"` and `role="region"` (which together convey the focusable "map component") are set on the `<canvas>` which means that any components that are placed outside of it (e.g. the `maplibregl-control-container` controls and potential custom UI components) aren't [_programmatically determinable_](https://www.w3.org/TR/WCAG21/#dfn-programmatically-determinable) as related to the map's [_structure_](https://www.w3.org/TR/WCAG21/#dfn-structure).

Notice how in the [_accessibility tree_](https://developer.mozilla.org/en-US/docs/Glossary/Accessibility_tree) the button is seemingly not part of any particular semantic structure:

![structure-current](https://user-images.githubusercontent.com/26493779/133896008-03f009e5-f1a8-4d7e-b8fe-d4207e08c2ab.png)

If `tabindex="0"`, `aria-label="Map"` and `role="region"` were moved from `<canvas>` to `<div class="maplibregl-map">` the correct structure would be accomplished (and logical focus order guaranteed). However, in this case I suppose the `div` would have to act as a proxy for `canvas` in terms of user input, and probably other things, which might get messy.

Another option is to use [`aria-owns`](https://www.w3.org/TR/wai-aria/#aria-owns) which is seemingly widely [supported](https://a11ysupport.io/tech/aria/aria-owns_attribute#expectations), straight forward to use, and produces the desired result:

![structure-aria-owns](https://user-images.githubusercontent.com/26493779/133896012-210e788f-7cd3-4174-9d5c-0bcb804a2723.png)

Although, the [number 1 rule of using ARIA](https://www.w3.org/TR/using-aria/#rule1) is to avoid using ARIA if you can. And there's also the consideration of how `aria-owns` should work with developers' custom UI components.

<hr>

Relates to <span title="Web Content Accessibility Guidelines">WCAG</span> 2.1 <span title="Success Criterion">SC</span> [1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships).

(The same issue was also reported with MapBox: https://github.com/mapbox/mapbox-gl-js/issues/11038.)

**Comment by HarelM** [comment]
Can you make a PR to solve all the issues you opened about accessibility?

**Comment by Malvoz** [comment]
@HarelM Unfortunately I am not familiar with the MapLibre codebase, and MapLibre is only 1 of [many web maps](https://github.com/Malvoz/web-maps-wcag-evaluation/blob/master/README.md) that I have reported issues for so it would not feasible for me to spend the time on all these projects.

**Comment by HarelM** [comment]
I'm not familiar with accessibility criteria and implementation. In the src folder there's the ui folder, in it all the components are, you can easily add this if this is important to you...
This is a community project, everyone helps where it's important to them, simply opening issues is less helpful than actually contributing code...

**Comment by wipfli** [comment]
Thanks @Malvoz for opening the accessibility issues. I think they are valuable and if questions regarding accessibility come up, for example like in https://github.com/maplibre/maplibre-gl-js/pull/344, I will take the freedom to ping you and ask for expert advice.

**Comment by HarelM** [comment]
I'm sorry, I didn't mean to discourage anyone, the reality is that there's no company to back this project and the best approach to progress something is to open PRs.
If this reality is discouraging I'm sorry...

**Comment by wipfli** [comment]
I think it is important to remember that MapLibre GL JS is really just a community-driven project. We all do this for fun and the goal is certainly not to discourage anyone from contributing.

**Comment by xabbu42** [comment]
@wipfli thanks and I completely agree.

@HarelM I was not discouraged by any reality outside of your control, but by your reaction in this issue, especially the following quote: 

> simply opening issues is less helpful than actually contributing code...

In that case I would actually disagree as accessibility experts are hard to come by and fixing this issue seems the smaller part of the work. But even if it were true, do you intend to mention that reality on every issue opened without a pull request? That behavior is discouraging to me, not the reality on how things progress in any open source project.

**Comment by HarelM** [comment]
I have deleted my comments. I will stop trying to help by pushing this project forward, reviewing every PR with valid comments and reviewing every issue that someone opens and try to direct them to action. My bad... I'm wondering who's discouraging who...

**Comment by thom4parisot** [comment]
It seems this issue has been closed but not adresssed although it is highly relevant, right?

I would be quite happy to tackle a few of them to have accessible maps on my websites.

Is there any funding available or shall I do that as part of my spare time?

**Comment by HarelM** [comment]
You are right, this wasn't addressed. Let me know if you would like me to reopen this issue. 
Currently maplibre organization does not have a way to fund things, but you can try contacting the board and ask for funding, it can't hurt I guess.

**Comment by thom4parisot** [comment]
Hi, thanks for your prompt reply 😄 

I'd be happy to contribute on my ~~spare time~~ work time.
I would like to tackle this issue, as well as a few coined by @Malvoz in #53 (thanks for your work by the way, it's a great place to start). Like #357, #358, #360, #361, #362, #363 and #11035.
